lua/luasnip/util/str.lua

```diff
@@ -113,4 +113,34 @@ function M.sanitize(str)
 	return str:gsub("%\r", "")
 end
 
+-- string-operations implemented according to
+-- https://github.com/microsoft/vscode/blob/71c221c532996c9976405f62bb888283c0cf6545/src/vs/editor/contrib/snippet/browser/snippetParser.ts#L372-L415
+-- such that they can be used for snippet-transformations in vscode-snippets.
+local function capitalize(str)
+	-- uppercase first character.
+	return str:sub(1, 1):upper() .. str:sub(2, -1)
+end
+M.vscode_string_modifiers = {
+	upcase = string.upper,
+	downcase = string.lower,
+	capitalize = capitalize,
+	pascalcase = function(str)
+		local pascalcased = ""
+		for word in str:gmatch("[a-zA-Z0-9]+") do
```

Comment:
Is not better to use `gsub` here?

Reply:
Ouh, yeah that would be better :+1:

Reply:
(alternatively, remove all non`[a-zA-Z0-9]`-characters with a second gsub, but I think intent is clearer with the for-version)